### PR TITLE
[jjo] support kube v1.16, reconcile kube.libsonnet with bitnami-labs/kube-libsonnet

### LINKED ISSUE #752: plans/timeline to support k8s 1.16
Hi!

GCP is starting to send out emails for GKE saying that 1.16 is going to be made available in the regular release channel, with automatic upgrades (if enabled) starting 2020-04-09.

1.16 appears to deprecate several APIs used for BKPR including `apps/v1beta1`, so... is work to support 1.16 underway? is there a timeline? is there something a k8s n00b like me can do to help?

Thanks.

### PR DESCRIPTION
**Pull request by jjo**
Fixes #752. 

* Update `kubeprod/` and `tests/` to v1.17 kube client libs,
  many vendor/ing changes
* Update `go` to 1.14
* Fix manifests/lib/kube.libsonnet:
  - fix apiVersions for v1.16+, supported since v1.14+
  - reconcile with bitnami-labs/kube-libsonnet#37
    targetting vendor/ing it with `jsonnet-bundle` in a followup PR
  - affinity support: remove `PodZoneAntiAffinityAnnotation()`,
    use existing `utils.weakNodeDiversity()` instead
* Update `Jenkinsfile` for v1.16 releases, only available in AKS and GKE
  preview (`rapid` channel) at the time of this PR
* Downgrade `oauth2-proxy` to v4.1.0 as v5.0.x/v5.1.x doesn't
  seem to work with aks-1.16, current oauth2-proxy master has
  some related commits (oauth2-proxy/oauth2-proxy#440), worth
  trying once we have v5.2 available

### CODE REVIEW

**Comment by dbarranco** [comment]
> Fix EKS issue, please

Which is the issue with EKS?
It just seems EKS with k8s 1.16 is not in GA yet:
https://github.com/aws/containers-roadmap/issues/487

File: Jenkinsfile [hunk 1]
```diff
@@ -698,7 +698,7 @@ spec:
                     }
 
                     // we use GKE for testing the generic platform
-                    def genericKversions = ["1.14"]
+                    def genericKversions = ["1.15"]
```
Comment (dbarranco) [inline Jenkinsfile:701]: What do you think of having several versions in the generic one?
For me sounds like `1.14`, `1.15`, `1.16` should be here

**Comment by jjo** [comment]
> I'd add a note explaining what weakNodeDiversity is it.

Ok, note I'm just using it as already existing (and used in other files)

> 
> Please, fix the EKS part and you can commit:
> 
> ```
> invalid version, supported values: 1.12, 1.13, 1.14
> ```

Depends on our internal jenkins-bkpr getting a newer `eksctl`, just merged that.

File: Jenkinsfile (hunk 1 above)
Comment (jjo) [reply Jenkinsfile:701]: > What do you think of having several versions in the generic one?
> For me sounds like `1.14`, `1.15`, `1.16` should be here

`1.16` not possible (uses GKE which doesn't yet have it available), re: `1.14` as this feature is experimental I wouldn't waste time in older kubernetes versions.

**Comment by jjo** [comment]
> Fix EKS issue, please

About to rebuild with updated `eksctl` in our pipeline.

**Comment by jjo** [comment]
> > Fix EKS issue, please
> 
> Which is the issue with EKS?
> It just seems EKS with k8s 1.16 is not in GA yet:
> [aws/containers-roadmap#487](https://github.com/aws/containers-roadmap/issues/487)

(?) indeed - Jenkinsfile has:
```
                    def eksKversions = ["1.14", "1.15"]
```
Currently 1.16 only test-able in AKS

**Comment by dbarranco** [comment]
> > > Fix EKS issue, please
> > 
> > 
> > Which is the issue with EKS?
> > It just seems EKS with k8s 1.16 is not in GA yet:
> > [aws/containers-roadmap#487](https://github.com/aws/containers-roadmap/issues/487)
> 
> (?) indeed - Jenkinsfile has:
> 
> ```
>                     def eksKversions = ["1.14", "1.15"]
> ```
> 
> Currently 1.16 only test-able in AKS, as per PR summary.

Oh ok, it's about the CI. Fine.
I thought Jorge was talking about EKS versions, no problem.

**Comment by jjo** [comment]
bors try

**Comment by jjo** [comment]
bors try

**Comment by jjo** [comment]
bors try

**Comment by jjo** [comment]
bors try

**Comment by dbarranco** [comment]
so it seems AKS succeeded in the last test execution but:

```
...
GitHub has been notified of this commit’s build result

ERROR: No test report files were found. Configuration error?
Finished: FAILURE
```

First time I see that one, maybe that's something related with a Jenkins plugin?

**Comment by jjo** [comment]
expanding to all platforms, forcing rebuild to catch Jenkinsfile parameters changes:

bors try

**Comment by jjo** [comment]
forcing rebuild to catch Jenkinsfile parameters changes:

bors try

**Comment by jjo** [comment]
bors try

**Comment by jjo** [comment]
\o/ two success-es in a row, triggering a try to pickup updated parameters (for 2 releases testing per cloud)

bors try

**Comment by jjo** [comment]
@jbianquetti-nami @dbarranco PTAL, tests passing Ok since some commits ago
https://jenkins-bkpr.nami.run/blue/organizations/jenkins/BKPR%2Fkube-prod-runtime/detail/pr-770/41/pipeline
jjo-kubeprod-pr-770

**Comment by jjo** [comment]
> Build failed:
> 
> * [continuous-integration/jenkins/branch](https://jenkins-bkpr.nami.run/job/BKPR/job/kube-prod-runtime/job/staging/128/display/redirect)

gke-1.15 tmp failure:
```
E0420 11:45:17.689280    2339 client.go:118] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
```

bors r+

**Pull request merged**